### PR TITLE
[150] Add a GitHub Action to run CI for PR's and pushes.

### LINKED ISSUE #150: Add CI server checks for PR's
Currently, only ip validation is performed for each request.
image

Having a CI integration to make sure the PR at least builds fine is something we can add to the project to automate test result for each PR.

The first approach can be https://travis-ci.org. 
What do you think?

### PR DESCRIPTION
**Pull request by jamezp**
resolves #150 

This should likely be merged after #368

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -17,12 +17,12 @@
...
     <modelVersion>4.0.0</modelVersion>
 
-   
+
     <parent>
         <groupId>org.eclipse.microprofile</groupId>
         <artifactId>microprofile-parent</artifactId>
-        <version>2.6</version>
-    </parent> 
+        <version>3.1</version>
```
Comment (Emily-Jiang) [inline pom.xml:24]:
```suggestion
        <version>3.2</version>
```

Comment (jamezp) [reply pom.xml:24]: Thank you, sorry I missed that. I've updated #368. We can either go with #368 first or I can combine them here. I'm okay either way.

Comment (Emily-Jiang) [reply pom.xml]: I approved the PR #368 . I think we can merge that one first and then you can rebase. What do you think?

Comment (jamezp) [reply pom.xml]: +1, once that is merged I'll get this one updated and it will only contain the workflow file.

Comment (Emily-Jiang) [reply pom.xml]: #368 was merged.

Comment (jamezp) [reply pom.xml]: Thank you! I've updated this commit and made it a PR.

**Review by Emily-Jiang — APPROVED** [review]
LGTM! Thanks @jamezp

**Pull request merged**